xdsl/dialects/riscv_snitch.py

```diff
@@ -129,8 +130,6 @@ class FRepOperation(IRDLOperation, RISCVInstruction):
     wraps again (up to 23 = 8).
     """
 
-    traits = frozenset((NoTerminator(),))
-
```

Comment:
I'm a bit confused by point 3 in your PR message;
why not use the `SingleBlockImplicitTerminator` trait like in `scf.for` and call `ensure_terminator` below?

Reply:
I somehow missed that trait, will add it now